### PR TITLE
weekly Renewal emails

### PR DESCRIPTION
**Pull request by pvighi**
Added confirmation email at the end of the guardian weekly renewal process
@paulbrown1982 @johnduffell

### CODE REVIEW

File: app/model/exactTarget/DataExtensionRow.scala [hunk 1]
```diff
@@ -262,6 +301,54 @@ object HolidaySuspensionBillingScheduleDataExtensionRow {
...
+    val commonFields = PaperFieldsGenerator.fieldsFor(
+      deliveryAddress = extractAddress(contact),
+      maybeBillingAddress = None,
+      includesDigipack = false,
+      email = email,
+      title = contact.title.flatMap(Title.fromString),
+      firstName = contact.firstName.getOrElse(""),
+      lastName = contact.lastName,
+      startDate= contractEffective,
+      firstPaymentDate = customerAcceptance,
```
Comment (pvighi) [inline app/model/exactTarget/DataExtensionRow.scala]: Are these two dates correct?

**Comment by johnduffell** [comment]
some more description would be good, mainly to help us in future when we're going back in time

File: app/services/CheckoutService.scala [hunk 2]
```diff
@@ -276,14 +278,18 @@ class CheckoutService(identityService: IdentityService,
     val ratePlan = RatePlan(renewal.plan.id.get, None, Nil)
     val amend =   Amend(subscriptionId = subscription.id.get, plansToRemove = Nil, newRatePlans = NonEmptyList(ratePlan))
 
+    val contractEffective = subscription.termEndDate
+    val customerAcceptance = nextFridayFrom(contractEffective)
```
Comment (paulbrown1982) [inline app/services/CheckoutService.scala]: Just a note that when we offer 'x issues free' renewal promotions, it's this customerAcceptance that will need extending by X-from-the-promotion days.

File: app/controllers/AccountManagement.scala [hunk 3]
```diff
@@ -294,17 +294,19 @@ object AccountManagement extends Controller with LazyLogging with CatalogProvide
     implicit val resolution: TouchpointBackend.Resolution = TouchpointBackend.forRequest(PreSigninTestCookie, request.cookies)
     implicit val tpBackend = resolution.backend
 
-    def validateRenewable(sub: Subscription[SubscriptionPlan.PaperPlan]) = if (sub.renewable) \/-(sub) else -\/("subscription is not renewable")
+    def validateRenewable(sub: Subscription[SubscriptionPlan.WeeklyPlan]) = if (sub.renewable) \/-(sub) else -\/("subscription is not renewable")
 
     def jsonError(message: String) = Json.toJson(Json.obj("errorMessage" -> message))
+    def description(sub:Subscription[SubscriptionPlan.WeeklyPlan], renewal: Renewal) = renewal.plan.charges.prettyPricing(sub.plan.charges.currencies.head)
```
Comment (johnduffell) [inline app/controllers/AccountManagement.scala]: watch your spacing: `sub: Subscription[SubscriptionPlan.WeeklyPlan]`

File: app/services/CheckoutService.scala [hunk 4]
```diff
@@ -276,14 +278,18 @@ class CheckoutService(identityService: IdentityService,
...
     def addPlan = {
-      val contractEffective = subscription.termEndDate
-      val customerAcceptance = nextFridayFrom(contractEffective)
       val newRatePlan = RatePlan(renewal.plan.id.get, None)
-      val renewCommand =  Renew(subscription.id.get, subscription.startDate, newRatePlan, contractEffective, customerAcceptance)
+      val renewCommand = Renew(subscription.id.get, subscription.startDate, newRatePlan, contractEffective, customerAcceptance)
       zuoraService.renewSubscription(renewCommand)
     }
 
+    def ensureEmail(contact:Contact) =  if (contact.email.isDefined) Future.successful(\/.right(())) else salesforceService.repo.addEmail(contact, renewal.email).map { either =>
```
Comment (johnduffell) [inline app/services/CheckoutService.scala]: 😆

Comment (johnduffell) [inline app/services/CheckoutService.scala]: `(contact:Contact)`

File: app/services/ExactTargetService.scala [hunk 5]
```diff
@@ -172,6 +173,48 @@ class ExactTargetService(
       }
     }
   }
+
+  def enqueueRenewalEmail(
+    oldSub:Subscription[Plan.WeeklyPlan],
```
Comment (johnduffell) [inline app/services/ExactTargetService.scala]: 🙈

File: app/services/ExactTargetService.scala [hunk 6]
```diff
@@ -172,6 +173,48 @@ class ExactTargetService(
...
+    renewal: Renewal,
+    subscriptionDetails: String,
+    contact: Contact,
+    email: String,
+    customerAcceptance: LocalDate,
+    contractEffective: LocalDate): Future[Unit] = {
+
+    sealed trait Error {
+      def msg : String
```
Comment (johnduffell) [inline app/services/ExactTargetService.scala]: 💀

File: conf/DEV.conf [hunk 7]
```diff
@@ -30,7 +30,6 @@ qa {
 
 aws.queue.welcome-email = "subs-welcome-email"
 aws.queue.holiday-suspension-email = "subs-holiday-suspension-email"
-
```
Comment (johnduffell) [inline conf/DEV.conf]: no need this change to whitespace

**Comment by johnduffell** [comment]
looks good to me 👍 🎉

File: app/services/ExactTargetService.scala [hunk 8]
```diff
@@ -172,6 +173,48 @@ class ExactTargetService(
...
+        email,
+        customerAcceptance,
+        contractEffective)
+      sendMessage <- EitherT(sendToQueue(row))
+    } yield (sendMessage)).run
+
+   sentMessage.map{
+        case \/-(sendMsgResult) => logger.info(s"Successfully enqueued ${oldSub.name.get}'s guardian weekly renewal email.")
+        case -\/(SimpleError(msg)) => logger.error(msg)
+        case -\/(ExceptionThrown(msg, ex)) => logger.error(msg, ex)
```
Comment (johnduffell) [inline app/services/ExactTargetService.scala]: do we need to add any context to these error messages? i.e. `.withTrace(s"email for ${oldSub.name.get})`

**Comment by johnduffell** [comment]
great, thanks! 👍 good idea keeping the error codes more centrally

**Comment by paulbrown1982** [comment]
👍 nice refactoring

**Pull request merged**